osu.Game/GameModes/Play/Player.cs

```diff
@@ -20,18 +22,37 @@ public class Player : OsuGameMode
         protected override BackgroundMode CreateBackground() => new BackgroundModeCustom(@"Backgrounds/bg4");
 
         public BeatmapInfo BeatmapInfo;
-        public Beatmap Beatmap;
+        public WorkingBeatmap Beatmap;
 
         public PlayMode PlayMode;
 
+        protected override IFrameBasedClock Clock => playerClock;
+
+        private InterpolatingFramedClock playerClock;
+        private IAdjustableClock sourceClock;
+
+        protected override void Dispose(bool isDisposing)
+        {
+            Beatmap?.Dispose();
+            base.Dispose(isDisposing);
+        }
+
+        protected override bool OnExiting(GameMode next)
+        {
+            //eagerly dispose as the finalizer runs too late right now.
+            Beatmap?.Dispose();

```

Comment:
"finalizer runs too late" ? Why is it relevant that the disposal happens earlier? AutoTrack related?